crates/telio-pq/src/proto.rs

```diff
@@ -203,20 +221,52 @@ async fn handshake(
     Ok(TunnelSock { tunn, sock })
 }
 
-pub fn parse_get_response(pkgbuf: &[u8]) -> super::Result<kyber768::Ciphertext> {
+pub fn parse_get_response(pkgbuf: &[u8], local_port: u16) -> io::Result<kyber768::Ciphertext> {
     let ip = Ipv4Packet::new(pkgbuf).ok_or(io::Error::new(
         io::ErrorKind::InvalidData,
         "Invalid PQ keys IP packet received",
     ))?;
 
+    validate_get_response_ip(&ip).map_err(|reason| {
+        io::Error::new(io::ErrorKind::InvalidData, format!("PQ keys IP: {reason}"))
+    })?;
+
     let udp = UdpPacket::new(ip.payload()).ok_or(io::Error::new(
         io::ErrorKind::InvalidData,
         "Invalid PQ keys UDP packet received",
     ))?;
 
+    validate_get_response_udp(&udp, local_port).map_err(|reason| {
+        io::Error::new(io::ErrorKind::InvalidData, format!("PQ keys UDP: {reason}"))
+    })?;
+
     parse_response_payload(udp.payload())
 }
 
+fn validate_get_response_ip(ip: &Ipv4Packet) -> Result<(), &'static str> {
+    if ip.get_source() != REMOTE_IP {
+        return Err("invalid src IP");
+    }
+    if ip.get_destination() != LOCAL_IP {
+        return Err("invalid dst IP");
+    }
+    if ip.get_next_level_protocol() != ip::IpNextHeaderProtocols::Udp {
+        return Err("invalid protocol");
+    }
```

Comment:
Could we additionally validate the checksum.

I believe it should be enough just to compare [chacksum()](https://docs.rs/pnet/latest/pnet/packet/ipv4/fn.checksum.html) against [get_checksum()](https://docs.rs/pnet/latest/pnet/packet/ipv4/struct.Ipv4Packet.html#method.get_checksum)

Reply:
Sure